openssl/src/pkey_ctx.rs

```diff
@@ -210,13 +210,22 @@ where
     #[inline]
     pub fn verify(&mut self, data: &[u8], sig: &[u8]) -> Result<bool, ErrorStack> {
         unsafe {
-            let r = cvt_n(ffi::EVP_PKEY_verify(
+            let r = ffi::EVP_PKEY_verify(
                 self.as_ptr(),
                 sig.as_ptr(),
                 sig.len(),
                 data.as_ptr(),
                 data.len(),
-            ))?;
+            );
+            // `EVP_PKEY_verify` is not terribly consistent about how it,
+            // reports errors. It does not clearly distinguish between 0 and
+            // -1, and may put errors on the stack in both cases.
+            // As a practical matter, there's no reasonable need for a caller
+            // to distinguish between an "error" (which often means things
+            // like failing to parse the DER bytes of the signature) and a
+            // signature verification, so treat them all the same.
+            ErrorStack::get();
```

Comment:
It feels a bit unfortunate to throw away the errors here. We could potentially return the error stack if nonempty and r == 0, and Ok(false) otherwise? Not sure if that'd break existing users though.

Reply:
as a practical matter, I think its basically impossible to use this function without handling both. I can switch to the impl you suggested easily enough.